cardano-api/src/Cardano/Api/Value.hs

```diff
@@ -366,11 +360,28 @@ instance FromJSON ValueNestedRep where
     where
       parsePid :: (Text, Aeson.Value) -> Parser ValueNestedBundle
       parsePid ("lovelace", q) = ValueNestedBundleAda <$> parseJSON q
-      parsePid (pid, q) =
-        case deserialiseFromRawBytesHex AsScriptHash (Text.encodeUtf8 pid) of
-          Just sHash -> ValueNestedBundle (PolicyId sHash) <$> parseJSON q
-          Nothing -> fail $ "Failure when deserialising PolicyId: "
-                         <> Text.unpack pid
+      parsePid (pid, quantityBundleJson) = do
+        sHash <-
+          note ("Failure when deserialising PolicyId: " <> Text.unpack pid) $
```

Comment:
"Failed to deserialise PolicyId: " would be a better error message.

Reply:
done